File: leetcode_ly/java/Tree/SumRootToLeafNumbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int sumNumbers(TreeNode root) {
        if(root == null) return 0;
        int total = getSum(root,0,0);
        return total;
    }
    
    public int getSum(TreeNode root,int sum,int total){
        if(root == null) return 0;
        sum = sum * 10 + root.val;
        if(root.left == null && root.right == null){
            total += sum;
        }
        return total + getSum(root.left,sum,total) + getSum(root.right,sum,total);
    }
}

File: leetcode_ly/c++/Tree/SumRootToLeafNumbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        return getResult(root,0);
    }
    
    int getResult(TreeNode* root,int sum){
        if(!root) return 0;
        if(!root->left && !root->right) return sum * 10 + root->val;
        else return getResult(root->left,sum * 10 + root->val) + getResult(root->right,sum * 10 + root->val);
    }
};